xarray/core/utils.py

```diff
@@ -36,6 +37,20 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _maybe_cast_to_cftimeindex(index):
+    from ..coding.cftimeindex import CFTimeIndex
+    from ..coding.times import _require_standalone_cftime
+
+    if not OPTIONS['enable_cftimeindex']:
+        return index
+    else:
+        _require_standalone_cftime()
+        try:
+            return CFTimeIndex(index)
```

Comment:
Rather than attempting to convert everything into a CFTimeIndex, this should at least check dtypes and only do so if `dtype=object`.